services.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+func build(app_url string, test_suite_url string, username string, access_key string) (string, error) {
+	if app_url == "" || test_suite_url == "" {
+		return "", errors.New(FILE_NOT_AVAILABLE_ERROR)
+	}
+
+	payload_values := createBuildPayload()
+	payload_values.App = app_url
+	payload_values.TestSuite = test_suite_url
+
+	payload, err := json.Marshal(payload_values)
+
+	final_payload := appendExtraCapabilities(string(payload))
+
+	// log.Print("Final payload -> ", string(final_payload))
+
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", BROWSERSTACK_DOMAIN+APP_AUTOMATE_BUILD_ENDPOINT, bytes.NewBuffer(final_payload))
+
+	req.SetBasicAuth(username+"-bitrise", access_key)
+
+	req.Header.Set("Content-Type", "application/json")
+
+	res, err := client.Do(req)
+
+	if err != nil {
+		return "", errors.New(fmt.Sprintf(HTTP_ERROR, err))
+	}
+
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		return "", errors.New(fmt.Sprintf(HTTP_ERROR, err))
+	}
+
+	return string(body), nil
+}
+
+// this function uploads both app and test suite
+func upload(app_path string, endpoint string, username string, access_key string) (string, error) {
+	if app_path == "" {
+		return "", errors.New(FILE_NOT_AVAILABLE_ERROR)
+	}
+
+	payload := &bytes.Buffer{}
+	multipart_writer := multipart.NewWriter(payload)
+	file, fileErr := os.Open(app_path)
+
+	defer file.Close()
+
+	// creates a new form data header
+	// reading and copying the file's content to form data
+	attached_file,
+		fileErr := multipart_writer.CreateFormFile("file", filepath.Base(app_path))
+
+	_, fileErr = io.Copy(attached_file, file)
+
+	if fileErr != nil {
+		return "", errors.New(FILE_NOT_AVAILABLE_ERROR)
+	}
+
+	err := multipart_writer.Close()
+
+	if err != nil {
+		return "", errors.New(INVALID_FILE_TYPE_ERROR)
+	}
+
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", BROWSERSTACK_DOMAIN+endpoint, payload)
+
+	req.SetBasicAuth(username, access_key)
```

Comment:
do we have to append -bitrise for username?
if yes? can we constant it for re-usability.

Reply:
resolved,
this is not currently supported in our existing structure, nor is it required, 
bitrise mentioned via build api is sufficient